src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -184,7 +184,7 @@ public override Iterator<TSource> Clone()
 
             public override void Dispose()
             {
-                if (_enumerator is IDisposable) ((IDisposable)_enumerator).Dispose();
+                ((IDisposable)_enumerator).Dispose();
```

Comment:
This will now blow up if _enumerator is null.  It should really be:
```C#
if (_enumerator != null)
    _enumerator.Dispose();
```
Note that no cast at all is needed.